lib/okta/directory_service.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+
+module Okta
+  class DirectoryService < Common::Client::Base
+    def get_apps
+      okta_service = Okta::Service.new
+      redis = Redis.new
+      base_url = Settings.oidc.base_api_url + '/api/v1/apps?limit=200&filter=status+eq+\"ACTIVE\"'
+      unfiltered_apps = recursively_get_apps(okta_service, base_url)
+
+      # matches any ISO date.
+      # rubocop:disable Metrics/LineLength
+      iso_pattern = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d)/
```

Comment:
Can this be a constant instead? ISO_PATTERN? and do a .freeze at the end?

Reply:
Done!